### PR TITLE
#498 Fix for throws low level exception

### PR DESCRIPTION
**Pull request by larsroettig**
see #498

### CODE REVIEW

File: app/code/Magento/Inventory/Exception/ProductIsNotAssignedToStockException.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+declare(strict_types=1);
+
+namespace Magento\Inventory\Exception;
+
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * @api
+ */
+class ProductIsNotAssignedToStockException extends LocalizedException
```
Comment (maghamed) [inline app/code/Magento/Inventory/Exception/ProductIsNotAssignedToStockException.php]: No Need to introduce new Exception Type, just use LocalizedException instead

**Review by maghamed — CHANGES_REQUESTED** [review]

File: app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php [hunk 2]
```diff
@@ -45,7 +46,15 @@ public function execute(string $sku, int $stockId): float
             ->where(ReservationInterface::STOCK_ID . ' = ?', $stockId)
             ->limit(1);
 
-        $reservationQty = $connection->fetchOne($select);
+        try {
+            $reservationQty = $connection->fetchOne($select);
+        } catch (\Exception $e) {
```
Comment (maghamed) [inline app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php]: looks like this is the wrong place for catching the exception 
As we don't have exceptional situation at the time of Reservation fetching

Comment (maghamed) [reply app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php]: The issue happens in 

 Magento\InventoryIndexer\Model\GetStockItemData

**Review by maghamed — CHANGES_REQUESTED** [review]

File: app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php (hunk 2 above)
Comment (larsroettig) [reply app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php]: The exception was thrown in past if the index was not build for the stock

**Pull request closed without merging**